test/unit_tests/arkode/C_serial/ark_test_forcingstep.c

```diff
@@ -0,0 +1,135 @@
+/* -----------------------------------------------------------------------------
+ * Programmer(s): Steven B. Roberts @ LLNL
+ * -----------------------------------------------------------------------------
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2002-2024, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * All rights reserved.
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ * -----------------------------------------------------------------------------
+ * Unit tests on several ODEs with analytical solutions to verify the
+ * ForcingStep module.
+ * ---------------------------------------------------------------------------*/
+
+#include <arkode/arkode_arkstep.h>
+#include <arkode/arkode_forcingstep.h>
+#include <nvector/nvector_serial.h>
+
+#if defined(SUNDIALS_EXTENDED_PRECISION)
+#define GSYM "Lg"
+#else
+#define GSYM "g"
+#endif
+
+/* RHS function for f_1(t, y) = t / y */
+static int f_forward_1(const sunrealtype t, const N_Vector y,
+                       const N_Vector ydot, void* const user_data)
+{
+  N_VInv(y, ydot);
+  N_VScale(t, ydot, ydot);
+  return 0;
+}
+
+/* RHS function for f_2(t, y) = 1 / y */
+static int f_forward_2(const sunrealtype t, const N_Vector y,
+                       const N_Vector ydot, void* const user_data)
+{
+  N_VInv(y, ydot);
+  return 0;
+}
+
+/* Integrates the ODE
+ * 
+ * y' = [t / y] - [1 / y]
```

Comment:
I don't understand how this is correct, since shouldn't the full RHS just be the sum of the two RHS functions above?
```suggestion
 * y' = [t / y] + [1 / y]
```